Zend/zend_opcode.c

```diff
@@ -1039,6 +1040,9 @@ ZEND_API int pass_two(zend_op_array *op_array)
 		opline++;
 	}
 
+	// TODO: Should we re-assign CE columns in opcache after loading them from cache?
+	tcc_assign_ce_columns();
```

Comment:
So yeah, this is a problem. Classes loaded from opcache may be immutable, which means that it's not possible to change the index. One could use a MAP pointer for this purpose, which adds an extra level of indirection.

Reply:
> CI failures are legit, seems there are various Segfaults and Bus errors.

I am looking into these. Thanks.

Reply:
> So yeah, this is a problem. Classes loaded from opcache may be immutable, which means that it's not possible to change the index. One could use a MAP pointer for this purpose, which adds an extra level of indirection.

Ah, I did not consider this. Classes can be shared between processes, so these processes cannot write their own stuff into them.

Reconsidering, I think I still need the class entries to have a consecutive integer ID that is unique. But unique for all classes that exist in opcache, which means that opcache should assign them and guarantee uniqueness. Sounds tricky. Then, each process could map that global ID to a column index in the local TCC. I'm not sure if that is what you mean by a MAP pointer though.